pennylane_lightning/core/src/gates/GateOperation.hpp

```diff
@@ -25,9 +25,9 @@ namespace Pennylane::Gates {
  * @brief Enum class for all gate operations
  */
 enum class GateOperation : uint32_t {
-    BEGIN = 0,
+    BEGIN = 1,
     /* Single-qubit gates */
-    Identity = 0,
+    Identity = 1,
```

Comment:
This has always bothered me -- I'd like to figure out how to get rid of this begin/end thing here and duplicate values, but for now I'll ask --- does this change cause any headaches for us, or is it overall beneficial?

Reply:
The amount of error messages popping after changing this to `1` was frightening at first and makes you want to change this back to `0` and forget about it. They do not appear in many places in the end. They are used in a few places to size the enum or iterate over it. 

> does this change cause any headaches for us, or is it overall beneficial?

I put that in following @AmintorDusko 's [comment](https://github.com/PennyLaneAI/pennylane-lightning/pull/485#discussion_r1309090301). I don't think it causes any headaches. It allowed us to catch a few lurking magic `0`'s, so I guess it makes the code more robust. The immediate benefit is I can add coverage to the `default` cases (`PL_ABORT`) for named gates and generators.

Reply:
Thank you for that, @vincentmr.